### PR TITLE
Fix ldap user sync missed email in email_address table

### LINKED ISSUE #18603: Inconsistent e-mail address sync from LDAP
### Gitea Version

1.16-rc1

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

Compiled from source.

### Database

_No response_

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Description

According to

https://github.com/go-gitea/gitea/issues/4497#issuecomment-459057141

LDAP sync cron task synchronizes only primary user mail; current sync code

https://github.com/go-gitea/gitea/blob/main/services/auth/source/ldap/source_sync.go#L157

updates only email column in user table and does not update extra e-mails nor is_primary state in this table.

There should be separate new method called there for consistent updating user primary e-mail (i.e. in users table and extra emails table with is_primary updating). LDAP should not add changed e-mail as new e-mail (like manual changing user email in admin panel does after https://github.com/go-gitea/gitea/pull/17549) but should update current primary e-mail only like it was working in gitea 1.13.

Same solution should be applied to LDAP sync code on user login if https://github.com/go-gitea/gitea/pull/18452#issuecomment-1029806729 is merged.

To be considered: separate storing user primary e-mail in two separate tables is error prone. It should be kept in only one place problably to avoid such bugs.

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by lunny**
Fix #18603 . Should we keep the old email address or delete it when sync user with new email from ldap?

### CODE REVIEW

**Comment by lafriks** [comment]
If old email address source is LDAP, it should be deleted

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
CI failure is related

**Comment by lunny** [comment]
> Details

We don't know that. LDAP user could add a new email address and change the primary email. So let it be and user could delete the old email manually.

**Review by silverwind — APPROVED** [review]

File: models/user/user.go [hunk 1]
```diff
@@ -860,15 +861,34 @@ func updateUser(ctx context.Context, u *User, changePrimaryEmail bool) error {
...
+		}
+
+		if !primaryEmailExist {
+			_, err = e.Insert(&EmailAddress{
+				Email:       u.Email,
+				UID:         u.ID,
+				IsActivated: true,
+				IsPrimary:   true,
+			})
+			return err
```
Comment (zeripath) [inline models/user/user.go:877]: this should be if err != nil {

Comment (lunny) [reply models/user/user.go:877]: Good catch

**Pull request merged**